core/src/main/scala/epollcat/internal/ch/SocketHelpers.scala

```diff
@@ -98,18 +106,26 @@ private[ch] object SocketHelpers {
   }
 
   def getLocalAddress(fd: CInt): SocketAddress = {
-    val addr = stackalloc[posix.netinet.in.sockaddr_in]()
+    val sz =
+      if (preferIPv4Stack)
+        sizeof[posix.netinet.in.sockaddr_in]
+      else
+        sizeof[posix.netinet.in.sockaddr_in6]
+    val addr = stackalloc[Byte](sz)
     val len = stackalloc[posix.sys.socket.socklen_t]()
-    !len = sizeof[posix.sys.socket.sockaddr].toUInt
+    !len = sz.toUInt
     if (posix
         .sys
         .socket
         .getsockname(fd, addr.asInstanceOf[Ptr[posix.sys.socket.sockaddr]], len) == -1)
       throw new IOException(s"getsockname: ${errno.errno}")
-    toInetSocketAddress(addr)
+    if (preferIPv4Stack)
+      toInet4SocketAddress(addr.asInstanceOf[Ptr[posix.netinet.in.sockaddr_in]])
+    else
+      toInet6SocketAddress(addr.asInstanceOf[Ptr[posix.netinet.in.sockaddr_in6]])
   }
 
-  def toInetSocketAddress(
+  def toInet4SocketAddress(
       addr: Ptr[posix.netinet.in.sockaddr_in]
   ): InetSocketAddress = {
     val port = posix.arpa.inet.htons(addr.sin_port).toInt
```

Comment:
Can you double check my thinking?  The port in the sockaddr_in passed in should already be in
network order. If it came from the OS, it is. If it was manually constructed, some blundered if it
is not.  

On some machines, htons() is actually a no-op.  I forget which architectures and often
have to either look at the system C header (where is is often a macro) or, more quickly,
write a C program for the architecture of interest.  I am saying this because this code
may be working (as a no-op) on some machines but may scramble the port on others.

Reply:
Somewhere around here there is  byte copy which, I think, I have seen done more simply.
Now that the 50,000 foot issues seem to be resolving, I can see if I can create a candidate
for a PR after this merges.

Reply:
I found the code that I had been looking for which might simplify this section.
I will do a design study after this merges to see if what I think will work actually
does.

Reply:
>  I am saying this because this code may be working (as a no-op) on some machines but may scramble the port on others.

Yikes 😂 I stumbled on the `htons` in various sources, such as here:
https://www.gta.ufrj.br/ensino/eel878/sockets/sockaddr_inman.html

Reply:
I think may "alert!" was at the proper places, but the wrong function.  I can check again
in the light, but it dawned on me that the port in the various Java structures is almost
surely host-order.  That would mean that a "network-order-to-host-order", aka ntohs()
would need to be done here.    

Somewhere, and I can track it down, some does a final htons() (host-to-network)
before filling the sockaddr.

The network-to-host-to-network trip seems messy but I think it is required by
the containers we are using.   Intel is a little-endian machine, network order
is big endian,  so both htons() and ntohs() do real work and are not noops.

Let me see if I can confirm my suspicions about the port being stored in
Java structures in host order.

Sorry for the jerking around.  This stuff is complicated.

Reply:
It is `getaddrinfo` that is doing the htons() under the covers.  That is why grepping the
file does not show that function in the code.   The port is being stored in the Java structures
as host order, which makes sense.

So, the fix here is, I believe, to use `ntohs()`.  Of course, this invites the question of why
it is working the way it is.

Reply:
I figured out _why_ htons() was working.  htons() and, the correction, ntohs() both swap
bytes, in the case of a 2-byte short they do exactly the same thing.  AB becomes BA in
both cases.  Of course, this is on little-endian systems.  On big endian systems, they
are both no-ops.    

I think it is worth using the right name ntohs() here to help readers follow along and
reduce the astonishment factor.

Not a huge accomplishment, but figuring out a "not knowing why?" is a good
way to start my day.